### PR TITLE
Refactor ProfileGranularityDropdown to be a stateless widget

### PR DESCRIPTION
**Pull request by kenzieschmoll**
no functional changes - just a cleanup

### CODE REVIEW

File: packages/devtools_app/lib/src/ui/vm_flag_widgets.dart [hunk 1]
```diff
@@ -14,24 +14,18 @@ import '../profiler/profile_granularity.dart';
 /// DropdownButton that controls the value of the 'profile_period' vm flag.
 ///
 /// This flag controls the rate at which the vm samples the CPU call stack.
-class ProfileGranularityDropdown extends StatefulWidget {
-  const ProfileGranularityDropdown(this.screenId);
+class ProfileGranularityDropdown extends StatelessWidget {
+  ProfileGranularityDropdown(this.screenId)
+      : profilerService = CpuProfilerService();
```
Comment (jacob314) [inline packages/devtools_app/lib/src/ui/vm_flag_widgets.dart:19]: creating a mutable profiler service in the constructor of a stateful widget seems like a bug. You don't want a new profiler service every time the widget rebuilds. Can you hang the service off the service manager, use a provider, or a stateful widget?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/ui/vm_flag_widgets.dart]: Looking back through the original code, the methods on CpuProfilerService probably could have / should have been written as static. All they really are are helpers that interact with the vm service. Cleaned this up - PTAL

File: packages/devtools_app/lib/src/profiler/cpu_profile_service.dart [hunk 2]
```diff
@@ -23,23 +22,16 @@ class CpuProfilerService {
...
-      serviceManager.vmFlagManager.flag(vm_flags.profilePeriod);
-
-  Future clearCpuSamples() {
+  static Future clearCpuSamples() {
     return serviceManager.service
         .clearCpuSamples(serviceManager.isolateManager.selectedIsolate.id);
   }
 
-  Future<dynamic> setProfilePeriod(String value) {
+  static Future<dynamic> setProfilePeriod(String value) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/profiler/cpu_profile_service.dart:30]: nit: make these extension methods on serviceManager.service?
I don't think we have any services implemented as static methods so this is a little confusing and hides the fact that you are depending on the current serviceManager.service.

**Review by jacob314 — APPROVED** [review]
lgtm

**Pull request merged**